documentation/final-deliverables/pdf-report.md

```diff
@@ -105,16 +105,55 @@ FR23. To "unmatch" Mentees and Mentors, the Mentee can communicate with the Ment
+- Credentials for interacting with the database are within firebase-init.js.
+- ‘index.js’ allows the Mentee and Mentor forms to be rendered independently from the central website.
+- ‘timeZones.json’ is utilized to populate the ‘time zone’ dropdown on the Mentee and Mentor forms.
+
+A major benefit to using React is the ability to use UI Libraries (we opted for Ant Design), as well as to integrate with our back-end Node server. The main Ant Design components utilized throughout the app are Notifications, Cards, Forms, Icons, Tabs, and Modals.
+
+**Technical Specifications on the Back-end:**
+
+- Node is a back-end environment which uses JavaScript.
+- Our back-end is accessed from the ‘back-end’ folder within ‘src’.
+- ‘App.js’ contains functions that are triggered from events on a specific port. For example, functions exist for generating matches between Mentees and Mentors (by finding the least difference in time zones, number of commons skills, and same organization string) as well as route calls for inviting new mentors and accepting mentees.
+- ‘Email-helpers.js’ houses functions that send custom emails to users via nodemailer.
+- The back-end is configured to use the Firebase Admin SDK as a way of securing user information.
+
+Note that both the front-end and back-end contain files to assist in catching errors (.eslintrc.json) and for automatic formatting adjustments (.prettierrc.json). The package-lock.json files contain the list of dependencies which are installed whenever the ‘npm install’ command is run in the terminal before running the front-end and back-end.
+
+**Technical Specifications of the Database:**
+
+- Firebase is a real-time NoSQL database owned by Google, which is hosted on the cloud.
+- By navigating to the ‘Realtime Database’ tab of the Firebase console, the JSON data can be referenced. Two nodes exist in our database’s structure:
+
+1.  Organizations - Allow multiple Admins to exist within an Organization, as well as contain mentor email addresses who are either active or pending (for clarity: activeMentors and pendingMentors are ‘children’ of the ‘Organization’ node).
+2.  Users – Contains unique user IDs for each type of user (Admin, Mentor, or Mentee), with the information (i.e ‘children’ of the ‘Users’ node) gathered from the form used to register.
+
+- The ‘Authentication’ tab of the Firebase console allows us to verify Mentor and Admin identities upon account creation as well as to validate credentials upon login. All users are required to verify their email address before being able to access Admin or Mentor privileges of the website.
+  - Our Firebase configuration uses the ‘Email/Password’ Sign-in method. This can be modified if needed in the future by going to the ‘Sign-in method’ tab under ‘Authentication’.
+  - Authentication also handles password reset changes.
```

Comment:
```suggestion
  - Authentication also handles password reset changes.
  - Since we use Firebase's 'Authentication' service, we never have to worry about encrypting or storing users' passwords, because that is also handled for us. 
```

Reply:
Ah, right! Thanks for adding this in. I remember asking about the password visibility back in October, so this is a really great point to incorporate